lib/Importer/ONNXModelLoader.cpp

```diff
@@ -1512,6 +1544,16 @@ Error ONNXModelLoader::loadPool(const ONNX_NAMESPACE::NodeProto &op,
       idx = AvgPoolNode::ResultIdx;
     }
     auto *N = G_->createTranspose(opName, NodeValue(node, idx), NHWC2NCHW);
+
+    // For maxPool1D
+    if (inDim == 3) {
+      auto *Nr = G_->createTranspose(opName, N, NCHW2NHWC);
+      auto *R = G_->createSqueeze(opName, Nr, 1);
+      auto *RR = G_->createTranspose(opName, R, {0, 2, 1});
+      RETURN_IF_ERR(addNodeAsOutput(op, RR));
+      return Error::success();
+    }
+
     RETURN_IF_ERR(addNodeAsOutput(op, N));
```

Comment:
nit: you're creating two back to back transposes which will get eliminated later. Might be cleaner to do something like:

```cpp
Node *N = nullptr;
if (inDim == 3) {
  auto *R = G_->createSqueeze(opName, NodeValue(node, idx), 1);
  N = G_->createTranspose(opName, R, {0, 2, 1})->getResult();
} else {
  N = G_->createTranspose(opName, NodeValue(node, idx), NHWC2NCHW)->getResult();
}
RETURN_IF_ERR(addNodeAsOutput(op, N));
```

Reply:
Done.